netket/optimizer/qgt/common.py

```diff
@@ -37,15 +38,50 @@ def check(x, target):
             vec_iscomplex = jnp.iscomplexobj(target)
 
         if not par_iscomplex and vec_iscomplex:
-            raise TypeError(
+            raise ComplexDomainError(
                 dedent(
                     """
                     Cannot multiply the (real part of the) QGT by a complex vector.
                     You should either take the real part of the vector, or perform
                     the multiplication against the real and imaginary part of the
                     vector separately and then recomposing the two.
+
+                    This is happening because you have real parameters or a non-holomorphic
+                    complex wavefunction. In this case, the Quantum Geometric Tensor is
+                    actually only the real part of it.
+
+                    If you were executing a matmul `G@vec`, try using:
+
+                       >>> vec_real = jax.tree_map(lambda x: x.real, vec)
+                       >>> G@vec_real
+
+                    or
+
+                       >>> vec_real = jax.tree_map(lambda x: x.real, vec)
+                       >>> G.solve(linear_solver, vec_real)
+
+                    to fix this error.
+
+                    NOTE!!!
+                    You probably should pay attention whever you need the real or imaginary part
+                    of the vector in your equations!
+
```

Comment:
Too many exclamation points...
```suggestion
                    Be careful whether you need the real or imaginary part
                    of the vector in your equations!
```